### PR TITLE
rgw/notifications: get/delete of specific notification

### PR DESCRIPTION
**Pull request by yuvalif**
note this is a fix to the tool and documentation not the code
(specific delete/get already supported in he code)

Fixes: https://tracker.ceph.com/issues/48369

Signed-off-by: Yuval Lifshitz <ylifshit@redhat.com>

## testing
* copy the new `service-2.sdk-extras.json` file into `~/.aws/models/s3/2006-03-01/`. if this directory does not exist already, create it first
* run a ceph cluster using vstart
* create a topic:
```
aws --endpoint-url http://localhost:8000 sns create-topic --name fishtopic --attributes='{"push-endpoint": "amqp://127.0.0.1:5672", "amqp-exchange": "ex1", "amqp-ack-level": "broker"}'
```
* create a bucket:
```
aws --endpoint-url http://localhost:8000 s3 mb s3://fish
```
* create 2 notifications:
```
aws --region=default --endpoint-url http://localhost:8000 s3api put-bucket-notification-configuration --bucket fish --notification-configuration='{"TopicConfigurations": [{"Id": "creation-notif", "TopicArn": "arn:aws:sns:default::fishtopic", "Events": ["s3:ObjectCreated:*"]}]}'

aws --region=default --endpoint-url http://localhost:8000 s3api put-bucket-notification-configuration --bucket fish --notification-configuration='{"TopicConfigurations": [{"Id": "deletion-notif", "TopicArn": "arn:aws:sns:default::fishtopic", "Events": ["s3:ObjectRemoved:*"]}]}'
```
* fetch **one** of the notifications using CLI:
```
aws --region=default --endpoint-url http://localhost:8000 s3api get-bucket-notification-configuration --bucket fish --notification "creation-notif"
```
* fetch all notifications using CLI:
```
aws --region=default --endpoint-url http://localhost:8000 s3api get-bucket-notification-configuration --bucket fish --notification ""
```
* delete **one** of the notifications using CLI (fetch all notifications to make sure only one is deleted):
```
aws --region=default --endpoint-url http://localhost:8000 s3api delete-bucket-notification-configuration --bucket fish --notification "creation-notif"
```
* delete all notifications using CLI  (fetch all notifications to make sure all are deleted):
```
aws --region=default --endpoint-url http://localhost:8000 s3api delete-bucket-notification-configuration --bucket fish --notification ""
```
* you can repeat the entire test (no need to create the topic and bucket again) with the `get_notification.py` and `delete_notification.py` boto3 scripts

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
this is fixing the aws cli but breaking the boto3 python API, when no notification is passed (the standard case)
more investigation is needed so that both options would work in python

**Comment by gauravsitlani** [comment]
@yuvalif I have tested the steps using aws cli , its working as expected.

**Comment by yuvalif** [comment]
> @yuvalif I have tested the steps using aws cli , its working as expected. Shall i repeat the tests with boto3 scripts as well ?

this would be great. thanks!

**Comment by mattbenjamin** [comment]
@gauravsitlani are you still working on this?

File: examples/boto3/README.md [hunk 1]
```diff
@@ -94,6 +94,21 @@ Expected output:
 }
 ```
 
+- Get configuration of all notification of a bucket:
+```
```
Comment (anthonyeleven) [inline examples/boto3/README.md:98]: "Get all notifications configured on a bucket:"

**Review by anthonyeleven — APPROVED** [review]

**Review by anthonyeleven — CHANGES_REQUESTED** [review]
See comments

**Pull request closed without merging**